designer/apps/pages/utils.py

```diff
@@ -0,0 +1,19 @@
+""" Page utils """
+
+
+def is_valid_child_page(cls, parent_child_pages):
+    """
+    When adding a child page (e.g., ProgramPage, EnterprisePage), we want to ensure
+    only child pages of the same type can exist under their parent page.
+
+    For example:
+    - if the parent page has no children, the page would be a valid sibling.
+    - if the parent page already has a child page `ProgramPage`, then only
+      a new `ProgramPage` is valid under the parent page.
+
+    .. no_pii:
+    """
+    for child_page in parent_child_pages:
+        if cls is not child_page.specific_class:
```

Comment:
Actually I would change:
```
if cls is not child_page.specific_class:
```
to
```
if not isinstance(child_page.specific_class, cls):
```

The `is` check is doing an identity check to see if they are the same object, whereas isinstance() actually compares class/subclass types (you'll see `if is not None` a lot cause there is only one None object in the python runtime)

This might work, but I think it's better form to use isinstance in this case 👍

Reply:
Assuming `cls` here is the class type and not an instance of a class

Reply:
@christopappas Thanks for the distinction here. I updated this line to use `isinstance` but with `child_page.specific` instead of `child_page.specific_class` (see [Wagtail documentation](http://docs.wagtail.io/en/v2.0/reference/pages/model_reference.html#wagtail.core.models.Page.specific) for more details).